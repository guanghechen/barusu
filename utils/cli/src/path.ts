import path from 'path'


/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetDir
 */
export function relativeOfWorkspace(
  workspace: string,
  targetDir?: null,
): undefined
export function relativeOfWorkspace(
  workspace: string,
  targetDir: string,
): string
export function relativeOfWorkspace(
  workspace: string,
  targetDir?: string | null,
): string | undefined
export function relativeOfWorkspace(
  workspace: string,
  targetDir?: string | null,
): string | undefined {
  const absoluteDir = absoluteOfWorkspace(workspace, targetDir)
  if (absoluteDir == null) return undefined
  const relativeDir: string = path.relative(workspace, absoluteDir)
  return path.normalize(relativeDir)
}


/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetDir
 */
export function absoluteOfWorkspace(
  workspace: string,
  targetDir?: null,
): undefined
export function absoluteOfWorkspace(
  workspace: string,
  targetDir: string,
): string
export function absoluteOfWorkspace(
  workspace: string,
  targetDir?: string | null,
): string | undefined
export function absoluteOfWorkspace(
  workspace: string,
  targetDir?: string | null,
): string | undefined {
  if (targetDir == null) return undefined
  const absoluteDir: string = path.resolve(workspace, targetDir)
  return absoluteDir
}
