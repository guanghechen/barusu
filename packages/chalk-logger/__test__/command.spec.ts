import path from 'path'
import {
  DEBUG,
  calcLoggerOptionsFromArgs,
  registerCommanderOptions,
} from '../src'


describe('command funcs', function () {
  test('generateOptions', function () {
    const logFilepath = path.resolve(__dirname, 'command.log')
    expect(
      calcLoggerOptionsFromArgs([
        '--log-level=debug',
        '--log-name=waw',
        '--log-mode=loose',
        '--log-flag=date',
        '--log-flag=no-inline',
        '--log-flag',
        'no-colorful',
        '--log-filepath',
        logFilepath,
        '--log-encoding=utf-8',
      ])
    ).toEqual({
      level: DEBUG,
      name: 'waw',
      mode: 'loose',
      date: true,
      inline: false,
      colorful: false,
      filepath: logFilepath,
      encoding: 'utf-8'
    })
  })

  test('registerCommanderOptions', function () {
    const infos: any[] = []

    const program = {
      option: (...args: any[]) => {
        infos.push(args)
        return program
      }
    }

    registerCommanderOptions(program)
    expect(infos).toMatchSnapshot('registered options')
  })
})
