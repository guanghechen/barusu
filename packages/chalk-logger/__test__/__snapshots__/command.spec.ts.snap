// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command funcs registerCommanderOptions: registered options 1`] = `
Array [
  Array [
    "--log-level <level>",
    "specify logger's level.",
  ],
  Array [
    "--log-name <name>",
    "specify logger's name.",
  ],
  Array [
    "--log-mode <'normal' | 'loose'>",
    "specify logger's name.",
  ],
  Array [
    "--log-flag <option>",
    "specify logger' option. [[no-]<date|colorful|inline>]",
    [Function],
    Array [],
  ],
  Array [
    "--log-filepath <filepath>",
    "specify logger' output path.",
  ],
  Array [
    "--log-encoding <encoding>",
    "specify output file encoding.",
  ],
]
`;
